src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java

```diff
@@ -0,0 +1,469 @@
+    private final ObjectProperty<GroupHierarchyType> groupHierarchySelectedProperty = new SimpleObjectProperty<>();
+
+    // Type
+    private final BooleanProperty typeExplicitProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeKeywordsProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeSearchProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeAutoProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeTexProperty = new SimpleBooleanProperty();
+
+    // Option Groups
+    private final StringProperty keywordGroupSearchTermProperty = new SimpleStringProperty("");
+    private final StringProperty keywordGroupSearchFieldProperty = new SimpleStringProperty("");
+    private final BooleanProperty keywordGroupCaseSensitiveProperty = new SimpleBooleanProperty();
+    private final BooleanProperty keywordGroupRegexProperty = new SimpleBooleanProperty();
+
+    private final StringProperty searchGroupSearchTermProperty = new SimpleStringProperty("");
+    private final BooleanProperty searchGroupCaseSensitiveProperty = new SimpleBooleanProperty();
+    private final BooleanProperty searchGroupRegexProperty = new SimpleBooleanProperty();
+
+    private final BooleanProperty autoGroupKeywordsOptionProperty = new SimpleBooleanProperty();
+    private final StringProperty autoGroupKeywordsFieldProperty = new SimpleStringProperty("");
+    private final StringProperty autoGroupKeywordsDelimiterProperty = new SimpleStringProperty("");
+    private final StringProperty autoGroupKeywordsHierarchicalDelimiterProperty = new SimpleStringProperty("");
+    private final BooleanProperty autoGroupPersonsOptionProperty = new SimpleBooleanProperty();
+    private final StringProperty autoGroupPersonsFieldProperty = new SimpleStringProperty("");
+
+    private final StringProperty texGroupFilePathProperty = new SimpleStringProperty("");
+
+    private Validator nameValidator;
+    private Validator nameContainsDelimiterValidator;
+    private Validator sameNameValidator;
+    private Validator keywordRegexValidator;
+    private Validator keywordSearchTermEmptyValidator;
+    private Validator searchRegexValidator;
+    private Validator searchSearchTermEmptyValidator;
+    private CompositeValidator validator = new CompositeValidator();
+
+    private final DialogService dialogService;
+    private final JabRefPreferences preferences;
+    private final BasePanel basePanel;
+    private final AbstractGroup editedGroup;
+
+    private final Character keywordDelimiter;
+
+    public GroupDialogViewModel(DialogService dialogService, BasePanel basePanel, JabRefPreferences preferences, AbstractGroup editedGroup) {
+        this.dialogService = dialogService;
+        this.preferences = preferences;
+        this.basePanel = basePanel;
+        this.editedGroup = editedGroup;
+
+        this.keywordDelimiter = Globals.prefs.getKeywordDelimiter();
```

Comment:
You already pass the JabRefPreferences as parameter

Reply:
Thanks. Forgot that in refactoring.